JavadocStyleGuide.md

```diff
@@ -0,0 +1,223 @@
+
+```java
+/**
+ * This &amp; that.
+ */
+```
+
+### 1.1.6 Code keywords, names, and variables
+
+Use `<code>...</code>` style for keywords and names including:
+- Java keywords
+- package names
+- class names
+- method names
+- interface names
+- field names
+- argument names
+- code examples (see section 1.3 below)
+
+```java
+/**
+ * Use the <code>Project</code> instance to configure the project.
+ */
+```
+
+### 1.1.7 Block tags
+
+Any of the standard "block tags" that are used appear in the order `@param`, `@return`, `@throw`s`, `@deprecated`, and these four types never appear with an empty description:
+
+```java
+/**
+ * @param  url  an absolute URL giving the base location of the image
+ * @param  name the location of the image, relative to the url argument
+ * @return      the image at the specified URL
+ * @see         Image
+ */
+```
+
+Full list of tags (in order):
+
+| # | Tag           | Usage                                                                          | Notes                                                                                      |
+|---|---------------|--------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------|
+| 1 | `@param`      | Methods and constructors only                                                  | Include this if applicable                                                                 |
+| 2 | `@return`     | Methods only                                                                   | Include this if applicable                                                                 |
+| 3 | `@exception`  | Same as `@throws`                                                              | Include this if applicable                                                                 |
+| 4 | `@see`        | Adds a “See Also” heading with a link or text entry that points to a reference | `@see string`<br>`@see <a href=”URL#value”>label</a>`<br>`@see package.class#member label` |
+| 5 | `@since`      | Adds a “Since” heading                                                         | Include the Gradle version if applicable                                                   |
+| 6 | `@deprecated` | Adds a comment indicating that this API should no longer be used               | Make sure to have an alternative API linked                                                |
+
+Custom Gradle tags:
+
+| # | Tag            | Usage                                        | Notes                                            |
+|---|----------------|----------------------------------------------|--------------------------------------------------|
+| 1 | `@apiNote`     | Adds a “API Note” heading                    | Describe interesting details about the API       |
+| 2 | `@implSpec`    | Adds a “Implementation Requirements” heading | Describe what we expect the plugin author to do  |
```

Comment:
❗ It's not just about plugin authors, I'd just say "require" and "implementing class".

Reply:
I think the "Notes" column should be renamed and expanded with more usage requirements. Some more background on these tags, that could be used here:

https://nipafx.dev/javadoc-tags-apiNote-implSpec-implNote/

`@apiSpec`
This is the one we know and love; a description that applies equally to all valid implementations of the method, including preconditions, postconditions, etc.

`@apiNote`
Commentary, rationale, or examples pertaining to the API.

`@implSpec`
This is where we say what it means to be a valid default implementation (or an overrideable implementation in a class), such as "throws UOE." Similarly this is where we'd describe what the default for `putIfAbsent` does. It is from this box that the would-be-implementer gets enough information to make a sensible decision as to whether or not to override.

`@implNote`
Informative notes about the implementation, such as performance characteristics that are specific to the implementation in this class in this JDK in this version, and might change. These things are allowed to vary across platforms, vendors and versions.

Reply:
Updated in the Style Guide. Thank you.